src/core/Icon/Icon.tsx

```diff
@@ -84,24 +84,15 @@ export class Icon extends Component<IconProps> {
     } = withSuomifiDefaultProps(this.props);
     const { className, ariaLabel } = this.props;
 
+    const iconColor =
+      color !== undefined ? color : colorValue({ tokens })('depthDark27');
+
     if (!!src) {
-      return (
-        <StyledIcon
-          src={src}
-          {...passProps}
-          color={colorValue({ tokens })('depthDark27')}
-        />
-      );
+      return <StyledIcon src={src} {...passProps} color={iconColor} />;
     }
 
     if (icon !== undefined) {
-      return (
-        <StyledSuomifiIcon
-          {...passProps}
-          icon={icon}
-          color={colorValue({ tokens })('depthDark27')}
-        />
-      );
+      return <StyledSuomifiIcon {...passProps} icon={icon} color={iconColor} />;
     }
 
     logger.warn(
```

Comment:
Not related to these changes, but I remember the policy being that we don't want nested ternary operators.

Reply:
I guess so, If you want to, you could try to write this in a better way?

Reply:
The easiest way would probably be to just pre-check a part of it and use that in the operator.
Something along the lines of:

const classNameErrorText = !!className ?  " with className: {className}" : '' ;

      `Icon ERROR${
        !!ariaLabel
          ? ` with aria-label: ${ariaLabel}`
          : classNameErrorText
      }`

Reply:
To write this in a different way would require for example a mutable data structure, a separate function with returns, or an if-elseif-else structure. => this format is shorter cleaner and easier to read. Won't change it unless explicit linter rules are applied to force this.